### PR TITLE
Forward  Button, made backbutton with abstract

### PR DESCRIPTION
**Pull request by m4rtint**
Almost Done.
If u want to look through the code for now.

### CODE REVIEW

File: Assets/Script/UI/MainMenu/HoldButton/HoldButton.cs [hunk 1]
```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using UnityEngine;
+using UnityEngine.UI;
+using TMPro;
+using InControl;
+
+public abstract class HoldButton : MonoBehaviour
```
Comment (m4rtint) [inline Assets/Script/UI/MainMenu/HoldButton/HoldButton.cs]: All Logic is in Abstract class. 

only need to replace some text and which button needs pressed

**Comment by m4rtint** [comment]
[EndGamePanel animation and button](https://trello.com/c/cSbJNvbb/259-endgamepanel-animation-and-button)

File: Assets/Script/Managers/WorldManagers/BaseWorldManager.cs [hunk 2]
```diff
@@ -49,8 +49,8 @@ private void Awake()
 
     private void Start()
     {
-        m_parkingManager.Initialize();
         m_playerManager.Initialize();
+        m_parkingManager.Initialize();
```
Comment (m4rtint) [inline Assets/Script/Managers/WorldManagers/BaseWorldManager.cs]: Player needs to be made, Then the parking spot is created and assigned to the player

ORDER MATTERS - DEPENDENCIES

**Comment by m4rtint** [comment]
[EndGamePanel animation and button](https://trello.com/c/cSbJNvbb/259-endgamepanel-animation-and-button)

File: Assets/Script/UI/MainMenu/HoldButton/HoldButton.cs [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    protected abstract string PlayStationButtonPrompt();
+
+    protected abstract string PCButtonPrompt();
+    
+    private string InitialButtonPrompt()
+    {
+        return UI.MainText.InitialHoldButtonMessage;
+    }
+
+    private void GetPCPromptForBackButton()
```
Comment (wbeltran777) [inline Assets/Script/UI/MainMenu/HoldButton/HoldButton.cs]: If we're abstracting, should these methods that talk about the Back Button be kept in the Back Button class?

Comment (m4rtint) [reply Assets/Script/UI/MainMenu/HoldButton/HoldButton.cs]: I misnamed it. 
Should be called GetPCPromptForButton()

Comment (m4rtint) [reply Assets/Script/UI/MainMenu/HoldButton/HoldButton.cs]: Fixed it

**Pull request merged**